src/lottie/lottieanimation.cpp

```diff
@@ -125,6 +140,26 @@ Surface AnimationImpl::render(size_t frameNo, const Surface &surface,
     return surface;
 }
 
+Surface AnimationImpl::renderPartial(size_t frameNo, const Surface &surface)
+{
+    bool renderInProgress = mRenderInProgress.load();
+    if (renderInProgress) {
```

Comment:
You can load bool without a local variable, can't you?
```suggestion
    if (mRenderInProgress.load()) {
```

Reply:
Moreover, to ensure the check works without concurrency conflicts from different threads you should call [std::atomic::exchange][1].

 [1]: https://en.cppreference.com/w/cpp/atomic/atomic/exchange

Reply:
Yes you can. However, you are commenting on code I haven't written (it's a copy and paste of the current render function below). So either we'd need to change all the render functions, or orthogonalize this part to its own common function. Please decide what's best and tell me.